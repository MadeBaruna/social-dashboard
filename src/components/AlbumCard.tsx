import React from 'react';
import { Card } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import ImageLoader from './ImageLoader';

interface IProps {
  id: number;
  title: string;
  thumbnail: string;
}

const AlbumCard = ({ id, title, thumbnail }: IProps) => (
  <Card as={Link} to={`/album/${id}`} fluid>
    <ImageLoader imageUrl={thumbnail} />
    <Card.Content>
      <Card.Description>{title}</Card.Description>
    </Card.Content>
  </Card>
);

export default AlbumCard;
