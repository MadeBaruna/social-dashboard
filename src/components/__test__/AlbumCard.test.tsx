import React from 'react';
import { shallow } from 'enzyme';
import AlbumCard from '../AlbumCard';

it('renders AlbumCard components', () => {
  const wrapper = shallow(<AlbumCard id={1} thumbnail="http://localhost/img.jpg" title="Title" />);
  expect(wrapper).toMatchSnapshot();
});
