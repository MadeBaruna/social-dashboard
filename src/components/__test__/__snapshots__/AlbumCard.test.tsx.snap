// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AlbumCard components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlbumCard
    id={1}
    thumbnail="http://localhost/img.jpg"
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": [Function],
      "children": Array [
        <ImageLoader
          imageUrl="http://localhost/img.jpg"
        />,
        <CardContent>
          <CardDescription>
            Title
          </CardDescription>
        </CardContent>,
      ],
      "fluid": true,
      "to": "/album/1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageUrl": "http://localhost/img.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <CardDescription>
            Title
          </CardDescription>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": [Function],
        "children": Array [
          <ImageLoader
            imageUrl="http://localhost/img.jpg"
          />,
          <CardContent>
            <CardDescription>
              Title
            </CardDescription>
          </CardContent>,
        ],
        "fluid": true,
        "to": "/album/1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageUrl": "http://localhost/img.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CardDescription>
              Title
            </CardDescription>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Title",
            },
            "ref": null,
            "rendered": "Title",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
